tasks/rh-sign-image/rh-sign-image.yaml

```diff
@@ -0,0 +1,88 @@
+---
+apiVersion: tekton.dev/v1
+kind: Task
+metadata:
+  name: rh-sign-image
+  labels:
+    app.kubernetes.io/version: "0.1.0"
+  annotations:
+    tekton.dev/pipelines.minVersion: "0.12.1"
+    tekton.dev/tags: release
+spec:
+  description: >-
+    Task to create internalrequests to sign snapshot components
+  params:
+    - name: snapshotPath
+      description: Path to the JSON string of the mapped Snapshot spec in the data workspace
+      type: string
+      default: "mapped_snapshot.json"
+    - name: dataPath
+      description: Path to the JSON string of the merged data to use in the data workspace
+      type: string
+      default: "data.json"
+    - name: requester
+      type: string
+      description: Name of the user that requested the signing, for auditing purposes
+    - name: requestTimeout
+      type: string
+      default: "180"
+      description: InternalRequest timeout
+  workspaces:
+    - name: data
+      description: workspace to read and save files
+  steps:
+    - name: sign-image
+      image: quay.io/hacbs-release/release-utils:a9603c0ae1982be5e1142d44765b1475adc7ad40
+      script: |
+        #!/usr/bin/env sh
+        #
+        SNAPSHOT_PATH=$(workspaces.data.path)/$(params.snapshotPath)
+        RUNDIR="$(workspaces.data.path)/$(context.taskRun.uid)"
+
+        if [ ! -d $RUNDIR ]; then
+          mkdir -p $RUNDIR
+        fi
+
+        DATA_FILE="$(workspaces.data.path)/$(params.dataPath)"
+        if [ ! -f "${DATA_FILE}" ] ; then
+            echo "No valid data file was provided."
+            exit 1
+        fi
+
+        default_pipeline_image="quay.io/redhat-isv/operator-pipelines-images:released"
+        request=$(jq -r '.sign.request // "hacbs-signing-pipeline"' ${DATA_FILE})
+        pipeline_image=$(jq -r --args default_pipeline_image "${default_pipeline_image}" \
+            '.sign.pipelineImage // "$default_pipeline_image"' ${DATA_FILE})
+        config_map_name=$(jq -r '.sign.configMapName // "signing-config-map"' ${DATA_FILE})
+        synchronously=$(jq -r '.sign.synchronously // "false"' ${DATA_FILE})
+
+        COMPONENTS_LENGTH=$(jq '.components |length' ${SNAPSHOT_PATH})
+        for (( COMPONENTS_INDEX=0; COMPONENTS_INDEX<COMPONENTS_LENGTH; COMPONENTS_INDEX++ )); do
+
+            reference=$(jq -r ".components[${COMPONENTS_INDEX}].containerImage" ${SNAPSHOT_PATH})
+            manifest_digest="${reference#*@}"
+
+            echo -en "Creating InternalRequest to sign image... "
+            internal-request -r "${request}" \
+                -p pipeline_image=${pipeline_image} \
+                -p reference=${reference} \
+                -p manifest_digest=${manifest_digest} \
+                -p requester=$(params.requester) \
+                -p config_map_name=${config_map_name} \
+                -t $(params.requestTimeout) -s ${synchronously} \
+                > ${RUNDIR}/ir-result.txt || \
```

Comment:
What if we did `| tee ${RUNDIR}/ir-result.txt` instead? That way all the output of the `internal-request` script will be printed and we don't need the `||` part at all.

Reply:
For now I will change as we won't implement the async for now. The difference is that in sync mode it exits according with the InternalRequest status and async it is always  0.

Reply:
Although, the sync mode will break the tests as we don't have the operator running.

Reply:
Why does the test need an operator? How does it fail? It should probably mock the `internal-request` call.

Reply:
Now it is mocking the `internal-request` and simulating the sync behavior.